src/bridge-deployer/src/commands/mod.rs

```diff
@@ -232,47 +272,64 @@ pub struct BFTArgs {
     controllers: Option<Vec<H160>>,
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub struct BftDeployedContracts {
     pub bft_bridge: H160,
     pub wrapped_token_deployer: H160,
+    pub fee_charge: H160,
+    pub minter_address: H160,
 }
 
 impl BFTArgs {
     /// Deploy the BFT contract
     pub async fn deploy_bft(
         &self,
-        network: EvmNetwork,
+        network: NetworkConfig,
         canister_id: Principal,
-        bridge: &Bridge,
         pk: H256,
         agent: &Agent,
+        is_wrapped_side: bool,
     ) -> anyhow::Result<BftDeployedContracts> {
         info!("Deploying BFT bridge");
 
         let contract_deployer = SolidityContractDeployer::new(network, pk);
 
-        let expected_nonce = contract_deployer.get_nonce().await? + 3;
+        let nonce_increment = match is_wrapped_side {
```

Comment:
Add a short description of why this is needed and what is the underlying logic

Reply:
This also seems fragile, if in the future we add another step before this one then the nonce will be different. Can it be replaced with an atomic counter passed as a parameter to the functions that need it?

Reply:
There is comment about that 2 lines below explaining this logic. I agree that this looks like a hack, but I don't know any other way to predict the nonce. We need to calculate the nonce before we deploy BftBridge contract, so before any operations are actually done.